### DISCUSSION TITLE
M5Stack Core S3

### DISCUSSION THREAD

**Discussion by andrzejroszkowski** [toverainc/willow#279] [General]
Hello,

Will M5Stack Core S3 (https://shop.m5stack.com/products/m5stack-cores3-esp32s3-lotdevelopment-kit) meet requirements for Willow?
If so, where to start with configuration?

Andrew

**Comment by kristiankielhofner** [comment]
We have sample units from M5Stack but there are some fairly significant hardware changes compared to the BOX line from Espressif. Our focus on new hardware support has been the new BOX-3. We won't have an exact ETA until we start looking at it in more depth but it's fairly straightforward in the grand scheme of things and will be coming "soon".

**Comment by andrzejroszkowski** [comment]
Pretty hard to find it "on stock"... is there any config to modify/adjust to fit it to M5 scheme?

**Reply by kristiankielhofner** [reply]
The BOX-3? Espressif only manufactured something like 250 of the pre-release version and yes those are all sold out.

We have been told by Espressif that a full production run with injection molded enclosure, etc will be available mid-October with a lot of stock.

It's far beyond config, from what I remember when we last looked at this they use (at least) a different LCD panel, etc that requires driver support in Willow. Low level and fundamental hardware changes like these are what I was referring to in terms of supporting it in Willow.

Unfortunately it's not as simple as installing a Windows driver or loading a Linux kernel module but that's how it is with $50 microcontroller based hardware.

**Reply by andrzejroszkowski** [reply]
true, performance over abstraction... will wait for BOX-3 then, in mean time, will take a look at code ;)

**Reply by kristiankielhofner** [reply]
If you want to take a look at support for the CoreS3 we would be really interested in a PR, any notes, etc!

Note that if you're going to look at support for it you should be targeting our [idf5.1 branch](https://github.com/toverainc/willow/tree/feature/idf5.1) which will be merged to main shortly.

**Comment by stintel** [comment]
I've had a deeper look at supporting the M5Stack CoreS3 over the weekend. Unfortunately it's going to be quite a bit of work to make it work. I committed my WIP code here:

https://github.com/toverainc/willow/tree/feature/m5stack_cores3
https://github.com/toverainc/esp-adf/tree/m5stack-cores3

If you want to have a go at adding support yourself, I suggest you look through those commits first. This is all very rough, and nothing actually works yet, aside from initializing the GPIO expander and toggling a GPIO pin there to power up the ES7210 chip.

**Reply by stintel** [reply]
@andrzejroszkowski I've spent some more time on the M5Stack CoreS3 last week, and updated the code in the branches I shared earlier. It's still very much WIP, but at this point we have display and microphone support, so it is actually possible to issue voice commands. There is no audio output yet, as the amplifier module is not supported by ESP-ADF, and the display backlight is always on.

We also noticed the device is substantially slower than the ESP32-S3-Box or ESP32-S3-Box-3, so if you do not have the hardware yet, we would recommend waiting for the Box-3.

**Reply by andrzejroszkowski** [reply]
will do :)

**Comment by mikey60** [comment]
AliExpress Espressif Store has 25 pieces of the ESP32-S3-BOX-3 left at the time of this post.

**Comment by andrzejroszkowski** [comment]
Order placed, delivery in T - ... Thanks for info :)
